src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/service/utils/ChangeManager.java

```diff
@@ -0,0 +1,46 @@
+package uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.service.utils;
+
+import java.util.function.BiConsumer;
+import java.util.function.BiPredicate;
+import java.util.function.Function;
+import uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.changes.benificialowner.Psc;
+
+public class ChangeManager {
```

Comment:
Not convinced by this way (using Generics) to do the comparisons.

Reply:
Why? This significantly reduces code reuse and will require a major refactor to change